include/deal.II/lac/vector.templates.h

```diff
@@ -259,14 +273,33 @@ namespace internal
       operator()(range.begin(),range.end());
     }
 #endif
-
-    void operator() (const size_type begin, const size_type end) const
+      void operator() (const size_type begin, const size_type end) const
     {
       DEAL_II_OPENMP_SIMD_PRAGMA
       for (size_type i=begin; i<end; ++i)
         val[i] += v_val[i];
     }
   };
```

Comment:
Why can't you just put an 'if (SIMD)' clause in the code up there? Since SIMD is a template parameter, there will be no cost in the check.